packages/reduce/module.d.ts

```diff
@@ -1,2 +1,12 @@
-export type Reducer<T> = (previous: T, current: T, index: number, arr: T[]) => T;
-export default function<T>(arr: T[], callback: Reducer<T>, initialValue?: T): T;
+export type Reducer<T, R> = (
```

Comment:
Can we make `R` default to `T`? In native `Array.reduce/Right` types, both values are `T`. BY having a default we also avoid breaking change.

```suggestion
export type Reducer<T, R=T> = (
```